packages/grpc-web/index.d.ts

```diff
@@ -70,11 +70,11 @@ declare module "grpc-web" {
 
   export class MethodDescriptor<REQ, RESP> {
     constructor(name: string,
-                methodType: any,
-                requestType: any,
-                responseType: any,
-                requestSerializeFn: any,
-                responseDeserializeFn: any);
+                methodType: string,
+                requestType: new (...args: unknown[]) => REQ,
+                responseType: new (...args: unknown[]) => RESP,
+                requestSerializeFn: (request: REQ) => Uint8Array,
+                responseDeserializeFn: (bytes: Uint8Array) => RESP);

```

Comment:
I am not quite sure why it would make sense to have these typed as wither `unknown` or `any` - if we allow more values here consumers will be allowed to create `MethodDescriptor` instances which break internal invariants, e.g. I could create a `MethodDescriptor` with a serialize function which returns a number and typescript won't complain, but during runtime it will break. In my opinion `MethodDescriptor` is part of the public interface and thus the constructor should only allow valid values to be passed to the constructor. Nevertheless I can totally understand if you want it to return an interface / unknown instead, maybe it's just a matter of personal preference.

Reply:
> if we allow more values here consumers will be allowed to create MethodDescriptor instances which break internal invariants

I get your sentiment in general.. But because MethodDescriptor code is generated rather than specified by callers, this probably doesn't make a practical difference..

> Nevertheless I can totally understand if you want it to return an interface / unknown instead, maybe it's just a matter of personal preference.

The main reason i want to keep this in sync with the Closure type annotation is that internally (Google) we're referring to the Closure type as source of truth, and only on Github we're also adding the TS type info. And because the internal types are subject to change, I don't want us to commit to stronger types in TS than we do in JS. If that makes sense :)

Reply:
Ok sure, thanks for the explanation. I reverted the types of `requestSerializeFn` and `responseDeserializeFn` to `any`.

Reply:
Thanks a lot :)